workflow/controller/operator.go

```diff
@@ -1360,7 +1370,11 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 
 	switch processedTmpl.GetType() {
 	case wfv1.TemplateTypeContainer:
-		node, err = woc.executeContainer(nodeName, templateScope, processedTmpl, orgTmpl, boundaryID)
+		if node != nil && node.Pending() {
+			_, err = woc.createWorkflowPod(node.Name, *processedTmpl.Container, processedTmpl, false)
+		} else {
```

Comment:
Could we move this logic inside of `woc.executeContainer` please? Or is there a reason as to why you kept it here?